tests/map_test.py

```diff
@@ -419,3 +419,72 @@ def test_pmap_unorderable():
 def test_supports_weakref():
     import weakref
     weakref.ref(m(a=1))
+
+
+class NonIterablePMap(PMap):
+    def __iter__(self):
+        assert False
+
+    def itervalues(self):
+        assert False
+
+    def values(self):
+        assert False
+
+    def iterkeys(self):
+        assert False
+
+    def keys(self):
+        assert False
+
+    def iteritems(self):
+        assert False
+
+    def items(self):
+        assert False
+
+
+def test_identity_equal_quick():
+    # It might be slow (O(n)) to iterate over all items in a PMap during
+    # comparison. If the maps happen to be the exact same object, there should
+    # be a quick path that does not require iterating over the PMap.
+
+    # Subclass used because PMaps use slots making them un-patchable.
+    m1 = NonIterablePMap(0, [])
```

Comment:
I had to go hunting the source code for the meaning of these arguments. Might be worth a comment.